kotskinds/apis/kots/v1beta1/helmchart_types.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2019 Replicated, Inc..
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type MappedChartValue struct {
+	Value string `json:",inline"`
+
+	valueType string `json:"-"`
+
+	strValue   string  `json:"-"`
+	boolValue  bool    `json:"-"`
+	floatValue float64 `json:"-"`
+
+	children       map[string]*MappedChartValue `json:"-"`
+	childrenArrays []*MappedChartValue          `json:"-"`
+}
+
+func (m *MappedChartValue) GetValue() interface{} {
+	if m.valueType == "string" {
+		return m.strValue
+	}
+	if m.valueType == "bool" {
+		return m.boolValue
+	}
+	if m.valueType == "float" {
+		return m.floatValue
+	}
+	if m.valueType == "nil" {
+		return nil
+	}
+
+	if m.valueType == "children" {
+		return m.children
+	}
+	if m.valueType == "childrenArray" {
+		return m.childrenArrays
+	}
+
+	panic("unknown value type")
```

Comment:
this function is called from `RenderValues`, which always returns nil error...  This should probably return an error and be handled there

Reply:
oh, and type info :)